### PR TITLE
dependency charts upgrade and misc upgrade improvement/fix

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
related issue: #3159, #2115

Bump several needed charts and misc fixes

### CODE REVIEW

**Review by guangbochen — COMMENTED** [review]
also need to patch the upgrade version check on https://github.com/harvester/harvester/blob/master/package/upgrade/upgrade_manifests.sh#L619, thanks.

**Comment by Vicente-Cheng** [comment]
> also need to patch the upgrade version check on https://github.com/harvester/harvester/blob/master/package/upgrade/upgrade_manifests.sh#L619, thanks.

sorry, I miss it.  Already update it, thanks!
Please help to review it again.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Review by starbops — APPROVED** [review]
Tested on v1.0.3 upgrade. LGTM, thanks!

**Comment by Vicente-Cheng** [comment]
Also, bump other charts like `harvester-network-controller`, `harvester-node-disk-manager`, and `harvester-node-manager`.
Please check it again. Thanks!

**Comment by bk201** [comment]
Should we update the node-manager's image to master-head or v0.1.3? https://github.com/harvester/harvester/blob/b0d883ce2173d3c485f5ef2dac652be9c6600dfa/deploy/charts/harvester/values.yaml#L467

**Comment by guangbochen** [comment]
> Should we update the node-manager's image to master-head or v0.1.3?
> 
> https://github.com/harvester/harvester/blob/b0d883ce2173d3c485f5ef2dac652be9c6600dfa/deploy/charts/harvester/values.yaml#L467

using `master-head` is preferred for the master branch, thanks.

**Comment by Vicente-Cheng** [comment]
> > Should we update the node-manager's image to master-head or v0.1.3?
> > https://github.com/harvester/harvester/blob/b0d883ce2173d3c485f5ef2dac652be9c6600dfa/deploy/charts/harvester/values.yaml#L467
> 
> using `master-head` is preferred for the master branch, thanks.

updated, thanks!

File: deploy/charts/harvester/values.yaml [hunk 1]
```diff
@@ -464,5 +464,5 @@ harvester-node-manager:
   image:
     imagePullPolicy: IfNotPresent
     repository: rancher/harvester-node-manager
-    tag: v0.1.2
+    tag: master-head
```
Comment (ibrokethecloud) [inline deploy/charts/harvester/values.yaml:467]: should the tag not be `v0.1.3`?

Comment (guangbochen) [reply deploy/charts/harvester/values.yaml:467]: we will tag it in the release branch, for the master branch prefer to use the current tag to make it consistent like others.

Comment (ibrokethecloud) [reply deploy/charts/harvester/values.yaml:467]: sorry yeah.. realized this is for master

**Review by ibrokethecloud — APPROVED** [review]
LGTM

**Review by starbops — APPROVED** [review]
Tested upgrade from v1.0.3 again and LGTM, thanks!

**Comment by bk201** [comment]
Try upgrading from v1.1.0. The final logs of apply_manifest:
```
hvst-upgrade-sgbjw-apply-manifests-mfc4b addon.harvesterhci.io/vm-import-controller patched                                                                                               
hvst-upgrade-sgbjw-apply-manifests-mfc4b addon.harvesterhci.io/pcidevices-controller patched
```

But addons don't seem to patch correctly:
```
$ k get addons.harvesterhci.io vm-import-controller -n harvester-system -o yaml
...
spec:
  chart: harvester-vm-import-controller
  enabled: true
  repo: http://harvester-cluster-repo.cattle-system.svc/charts
  valuesContent: |
    image:
      tag: v0.1.1
    fullnameOverride: harvester-vm-import-controller
  version: 0.1.2
status:
  status: AddonDeploySuccessful

$ k get addons.harvesterhci.io pcidevices-controller -n harvester-system -o yaml
...
spec:
  chart: harvester-pcidevices-controller
  enabled: false
  repo: http://harvester-cluster-repo.cattle-system.svc/charts
  valuesContent: |
    image:
      tag: v0.2.2
    fullnameOverride: harvester-pcidevices-controller
  version: 0.2.3
```

The add-on versions look OK, but the `image.tag` doesn't change. @ibrokethecloud Any takes to this?

**Comment by bk201** [comment]
Looks like only version is patched: https://github.com/harvester/harvester/blob/b0d883ce2173d3c485f5ef2dac652be9c6600dfa/package/upgrade/lib.sh#L256-L259

**Comment by Vicente-Cheng** [comment]
Hi all,
I update this PR, including the dependency charts upgrade and the grub issue on the upgrade path.
Please help to review it again. Thanks!!

**Review by ibrokethecloud — APPROVED** [review]
LGTM.

Tried out an upgrade, and can see the addons are patched, with both `version` and `valuesContent` getting updated.

```
Logging Event Audit: nothing to do in v1.1.0
Applying extra manifests
Warning: resource settings/default-data-path is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
setting.longhorn.io/default-data-path configured
Warning: resource customresourcedefinitions/ippools.whereabouts.cni.cncf.io is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
customresourcedefinition.apiextensions.k8s.io/ippools.whereabouts.cni.cncf.io configured
Warning: resource customresourcedefinitions/overlappingrangeipreservations.whereabouts.cni.cncf.io is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
customresourcedefinition.apiextensions.k8s.io/overlappingrangeipreservations.whereabouts.cni.cncf.io configured
addon.harvesterhci.io/vm-import-controller patched
addon.harvesterhci.io/pcidevices-controller patched

demo:~ # kubectl get addons.harvesterhci -o yaml -n harvester-system
apiVersion: v1
items:
- apiVersion: harvesterhci.io/v1beta1
  kind: Addon
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"harvesterhci.io/v1beta1","kind":"Addon","metadata":{"annotations":{},"labels":{"addon.harvesterhci.io/experimental":"true"},"name":"pcidevices-controller","namespace":"harvester-system"},"spec":{"chart":"harvester-pcidevices-controller","enabled":false,"repo":"http://harvester-cluster-repo.cattle-system.svc/charts","valuesContent":"image:\n  tag: v0.2.2\nfullnameOverride: harvester-pcidevices-controller\n","version":"0.2.2"}}
    creationTimestamp: "2022-11-25T05:22:47Z"
    generation: 2
    labels:
      addon.harvesterhci.io/experimental: "true"
    name: pcidevices-controller
    namespace: harvester-system
    resourceVersion: "85398"
    uid: 59c1c0c5-acd2-45e7-a2f1-77e30f0a9305
  spec:
    chart: harvester-pcidevices-controller
    enabled: false
    repo: http://harvester-cluster-repo.cattle-system.svc/charts
    valuesContent: |
      image:
        tag: v0.2.3
      fullnameOverride: harvester-pcidevices-controller
    version: 0.2.3
- apiVersion: harvesterhci.io/v1beta1
  kind: Addon
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"harvesterhci.io/v1beta1","kind":"Addon","metadata":{"annotations":{},"name":"vm-import-controller","namespace":"harvester-system"},"spec":{"chart":"harvester-vm-import-controller","enabled":false,"repo":"http://harvester-cluster-repo.cattle-system.svc/charts","valuesContent":"image:\n  tag: v0.1.1\nfullnameOverride: harvester-vm-import-controller\n","version":"0.1.1"}}
    creationTimestamp: "2022-11-25T05:22:47Z"
    generation: 2
    name: vm-import-controller
    namespace: harvester-system
    resourceVersion: "85390"
    uid: 5cd68687-1896-4ea1-9eed-fac455b2a168
  spec:
    chart: harvester-vm-import-controller
    enabled: false
    repo: http://harvester-cluster-repo.cattle-system.svc/charts
    valuesContent: |-
      image:
        tag: v0.1.2
      fullnameOverride: harvester-vm-import-controller
    version: 0.1.2
kind: List
metadata:
  resourceVersion: ""
```

**Review by bk201 — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]
LGTM and verified that the upgrade works, thanks.

**Review by starbops — APPROVED** [review]
LGTM, thanks!

**Pull request merged**